### PR TITLE
C350541: Recall an Item by Placing a Title-level Request Using Patron Services (MOD-PATRON): Item checked out with fixed due date

### PR DESCRIPTION
**Pull request by kimkate303**
https://foliotest.testrail.io/index.php?/cases/view/350541

### CODE REVIEW

File: cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import Checkout from '../../../support/fragments/checkout/checkout';
+import CheckInActions from '../../../support/fragments/check-in-actions/checkInActions';
+import Requests from '../../../support/fragments/requests/requests';
+import NewRequest from '../../../support/fragments/requests/newRequest';
+import InventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
+import FixedDueDateSchedules from '../../../support/fragments/circulation/fixedDueDateSchedules';
+import LoanPolicyActions from '../../../support/fragments/circulation/loan-policy';
+
+describe('ui-requests: Request: Edit requests. Make sure that edits are being saved.', () => {
+  // TODO: A client configured to use edge-patron API
```
Comment (penfriend) [inline cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js:23]: I have some doubts that we need to test here using UI. There is a diagram that shows that we need to call the backend module mod-patron directly. We need to clarify this asking the PO.
https://wiki.folio.org/display/FOLIOtips/Edge+APIs?preview=/25723203/25723207/EdgePatron.png#EdgeAPIs-Diagram.1

**Review by penfriend — CHANGES_REQUESTED** [review]

File: cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import CirculationRules from '../../../support/fragments/circulation/circulation-rules';
+import Checkout from '../../../support/fragments/checkout/checkout';
+import CheckInActions from '../../../support/fragments/check-in-actions/checkInActions';
+import Requests from '../../../support/fragments/requests/requests';
+import NewRequest from '../../../support/fragments/requests/newRequest';
+import InventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
+import FixedDueDateSchedules from '../../../support/fragments/circulation/fixedDueDateSchedules';
+import LoanPolicyActions from '../../../support/fragments/circulation/loan-policy';
+
+describe('ui-requests: Request: Edit requests. Make sure that edits are being saved.', () => {
```
Comment (penfriend) [inline cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js:22]: Please clarify why you add such description : Request: Edit requests. Make sure that edits are being saved.? Is it related to the test case?

**Comment by penfriend** [comment]
Please add the ability to have an option of TLR request and select this option during the creation of the TLR request

File: cypress.json [hunk 3]
```diff
@@ -16,4 +16,4 @@
     "grepFilterSpecs": true,
     "grepOmitFiltered": true
   }
-}
+}
```
Comment (IhorBohdan) [inline cypress.json:19]: Need this change?

File: cypress/integration/requests/patron-initiated/request-item-with-rolling-loan-recall.spec.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      firstName: userData1.personal.firstName,
+      middleName: userData1.personal.middleName,
+    }
+  };
+
+  before('Creating circ rule with request policy, users and item with rolling loan period', () => {
+    cy.getAdminToken();
+
+    // Create request policy that allows recalls and pages
+    RequestPolicy.createApi(requestPolicyWithRecall).then((body) => {
```
Comment (IhorBohdan) [inline cypress/integration/requests/patron-initiated/request-item-with-rolling-loan-recall.spec.js:55]: Maybe better rename to createViaApi

File: cypress.json (hunk 3 above)
Comment (kimkate303) [reply cypress.json:19]: esLint magic. opened file itself, looks similar to the previuos version

File: cypress/integration/requests/patron-initiated/request-item-with-rolling-loan-recall.spec.js (hunk 4 above)
Comment (kimkate303) [reply cypress/integration/requests/patron-initiated/request-item-with-rolling-loan-recall.spec.js]: done)

File: cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    // Create request policy that allows recalls and pages
+    RequestPolicy.createViaApi(requestPolicyWithRecall).then((body) => {
+      testData.requestPolicy = body;
+    });
+
+    // Create loan policy
+    FixedDueDateSchedules.createViaApi()
+      .then((schedule) => {
+        testData.scheduleId = schedule.id;
+        LoanPolicyActions.createApi(LoanPolicyActions.getDefaultLoanPolicy(testData.limitOfItem, schedule.id))
```
Comment (IhorBohdan) [inline cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js:66]: rename to createViaApi

Comment (kimkate303) [reply cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js:66]: i see, fine

File: cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+      });
+
+    // Create a circulation rule with an associated request policy that allows recalls and pages and with loan policy that uses fixed due date periods
+    cy.getCirculationRules().then((res) => {
+      testData.baseRules = res.rulesAsText;
+      testData.ruleProps = CirculationRules.getRuleProps(res.rulesAsText);
+      testData.ruleProps.r = testData.requestPolicy.id;
+      testData.ruleProps.l = testData.loanPolicy.id;
+
+      CirculationRules.addRuleApi(res.rulesAsText, testData.ruleProps, 'g ', testData.patronGroup.id).then(() => {
```
Comment (IhorBohdan) [inline cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js:107]: rename to ViaApi

File: cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        .then(requestId => {
+          Requests.deleteRequestApi(requestId);
+          Users.deleteViaApi(userData0.id);
+          Users.deleteViaApi(userData1.id);
+          PatronGroups.deleteViaApi(testData.patronGroup.id);
+        });
+    });
+
+    InventoryInstances.deleteInstanceAndHoldingRecordAndAllItemsViaApi(item.itemBarcode);
+    CirculationRules.deleteRuleApi(testData.baseRules);
```
Comment (IhorBohdan) [inline cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js:134]: rename to ViaApi

File: cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    RequestPolicy.deleteApi(testData.requestPolicy.id);
```
Comment (IhorBohdan) [inline cypress/integration/requests/patron-initiated/request-item-with-fixed-loan-recall.spec.js:135]: rename to ViaApi

**Review by IhorBohdan — APPROVED** [review]

**Pull request closed without merging**